src/core/simulation.jl

```diff
@@ -840,21 +840,17 @@ function update_parameter!(
     horizon = length(model_time_steps(problem.internal.optimization_container))
     for d in components
         # RECORDER TODO: Parameter Update from forecast
-        # TODO: Improve file read performance
-        forecast = PSY.get_time_series(
-            PSY.Deterministic,
-            d,
-            get_data_label(param_reference);
-            start_time = initial_forecast_time,
-            count = 1,
-        )
-        ts_vector = IS.get_time_series_values(
-            d,
-            forecast,
-            initial_forecast_time;
-            len = horizon,
-            ignore_scaling_factors = true,
-        )
+        TimerOutputs.@timeit RUN_SIMULATION_TIMER "GetTimeSeries" begin
```

Comment:
Can we remove this @timeit from here please? This operation happens way too often to record individually for every component. If anything, it might be delaying the loop. We can add the recorder too.